sqlmesh/core/snapshot/definition.py

```diff
@@ -927,6 +935,16 @@ def to_end_date(end: int, unit: IntervalUnit) -> TimeLike:
             else default
         )
 
+    def do_destructive_check(
+        self, allow_destructive_snapshots: t.Set[str], check_forward_change: bool = False
+    ) -> bool:
+        return (
+            self.is_model
+            and not self.model.on_destructive_change.is_ignore
+            and self.name not in allow_destructive_snapshots
+            and (self.is_forward_only if check_forward_change else True)
```

Comment:
Why do we care about `check_forward_change`?

Reply:
A snapshot needs to be checked if either:
1. It's a forward-only model with on_destructive_change != ignore
2. It's a model in a forward-only plan with on_destructive_change != ignore

We determine 1 in both the plantime and runtime checks.

We can determine 2 In the plantime check because we know if we're in a forward-only plan. 

In the runtime check, we do not know if we're In a forward-only plan but can infer 2 if change==FORWARD_ONLY. The `check_forward_change` arg lets us use this method for both plantime and runtime checks.